### DISCUSSION TITLE
FATAL is package manager is yarn v3

### DISCUSSION THREAD

**Discussion by tisonkun** [hexojs/hexo#5035] [Question / Answer] [answered]
You can checkout this version of repo: https://github.com/tisonkun/tisonkun.github.io/tree/3e4cb0a85a8e3b6c357c5bad04583027fd207fcf

Run `hexo s` with corepack or yarn v3 install, then it will fail with `FATAL`.

**Comment by stevenjoezhang** [comment, marked as answer]
See https://github.com/hexojs/hexo-cli/pull/227

**Reply by tisonkun** [reply]
@stevenjoezhang thanks for your reply! Does it mean that hexo doesn't plan to support yarn 2 and yarn 3?

**Reply by stevenjoezhang** [reply]
yarn 3 is currently not supported because Hexo relies on `node_modules` to manage plugins, which doesn't work the same way yarn 3 does. Maybe in the future we can find a way to support yarn 3, but for now it is recommended to use npm to install the hexo plugin.

**Reply by tisonkun** [reply]
Thank you!

**Reply by tisonkun** [reply]
@stevenjoezhang IIRC yarn 3 now supports `node_modules` resolution: https://yarnpkg.com/configuration/yarnrc#nodeLinker

Maybe you can review this decision again.

**Reply by stevenjoezhang** [reply]
If the package manager supports `node_modules` resolution, then the Hexo plugin system should work fine without modifying the source code. You can try yarn3 to see if it is compatible with Hexo now.

**Reply by tisonkun** [reply]
@stevenjoezhang It works. With once warning (failure?):

```
➤ YN0012: │ hexo-generator-archive@npm:2.0.0 isn't supported by any available linker
```

You can see https://github.com/tisonkun/tisonkun.github.io/pull/56 as a reference.

Locally I can run `hexo s` and preview correctly.

**Reply by tisonkun** [reply]
It seems a typo and now I find a way to make thing work :)
